src/typing/flow_js.ml

```diff
@@ -3644,6 +3644,15 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       in
       rec_flow cx trace (o, u)
 
+    | (NumT (reason_x, Literal (n, _)), ElemT(reason_op, (ObjT _ as o), t, rw)) ->
+      let x = string_of_float_trunc n in
```

Comment:
The problem with `string_of_float_trunc` is that it would fold distinct properties together, no? 
```
> let obj = {}
< Object {}
> obj[1] = "hey"
< "hey"
> obj[1.1] = 999
< 999
```